server/controllers/admin.py

```diff
@@ -859,6 +859,21 @@ def clients():
 
     return render_template('staff/clients.html', clients=clients, form=form)
 
+@admin.route("/clients/<string:client_id>", methods=['GET', 'POST'])
+@is_staff()
+def client(client_id):
+    client = Client.query.get(client_id)
+    client.client_secret = utils.generate_secret_key()
+    form = forms.ClientForm(obj=client)
+    if form.validate_on_submit():
+        client = Client(user=current_user)
```

Comment:
We shouldn't create a new client here, instead just use the one returned from the query above